dttools/src/md5.c

```diff
@@ -406,46 +423,31 @@ const char * md5_dir_rec(char * src){
 						return 0;
 					}
 				}
+			free(path);
 		}
 		closedir(dir);
 		return hash;
 	}
-	else if(S_ISREG(info.st_mode)){
```

Comment:
I think this removal is incorrect. The branches should stay the same (if/else).

The only part I suggested to move before the if is: `unsigned char digest[MD5_DIGEST_LENGTH];`